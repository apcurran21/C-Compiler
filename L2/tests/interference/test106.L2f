(@printArray
	1
	%array <- rdi
	%arraySize <- 2
	mem rsp -8 <- :checkArray_ret
	call @checkArray 1
	:checkArray_ret
	cjump rax = 0 :isInt1
	goto :isAnArray1
	:isInt1
	rax <- 0
	return
	:isAnArray1
	%i <- 0
	cjump %i < %arraySize :loop_start
	goto :loop_end
	:loop_start
	rdi <- %array
	rsi <- %i
	mem rsp -8 <- :printArrayElement_ret
	call @printArrayElement 2
	:printArrayElement_ret
	%i++
	cjump %i < %arraySize :loop_start
	goto :loop_end
	:loop_end
	return
)
