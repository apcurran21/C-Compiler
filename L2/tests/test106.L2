(@main
  (@firstNPrimes
    1
    %n <- rdi
    %isLessThanZero <- %n < 1

    cjump %isLessThanZero = 1 :call_label0
    goto :call_label1
    :call_label0
    rax <- -1
    return
    :call_label1
    rdi <- %n
    rsi <- 1
    call allocate 2
    %arr <- rax
    %n <- %n
    %n -= 1
    %n <- %n
    %n <<= 1
    %numPrimes <- 0
    %i <- 2
    %offset <- 8
    :call_label2
    %isNotDone <- %numPrimes < %n

    cjump %isNotDone = 1 :call_label3
    goto :call_label6
    :call_label3
    %encodedI <- %i
    %encodedI <<= 1
    %encodedI <- %encodedI
    %encodedI += 1
    rdi <- %encodedI

mem rsp -8 <- :call_label7
    call @isPrime 1

:call_label7
    %numIsPrime <- rax
    %numIsPrime <- %numIsPrime
    %numIsPrime -= 1
    %numIsPrime <- %numIsPrime
    %numIsPrime >>= 1
    cjump %numIsPrime = 1 :call_label4
    goto :call_label5
    :call_label4
    %address <- %arr
    %address += %offset
    mem %address 0 <- %encodedI
    %offset <- %offset
    %offset += 8
    %numPrimes <- %numPrimes
    %numPrimes += 1
    :call_label5
    %i <- %i
    %i += 1
    goto :call_label2
    :call_label6
    rax <- %arr
    return
  )
  (@isPrime
    1
    %n <- rdi
    %n <- %n
    %n -= 1
    %n <- %n
    %n >>= 1
    %isNLessThan2 <- %n < 2

    cjump %isNLessThan2 = 1 :call_label_ciao_8
    goto :call_label_ciao_9
    :call_label_ciao_8
    rax <- 1
    return
    :call_label_ciao_9
    %i <- 2
    %end <- %n
    %end -= 1
    :call_label_ciao_10
    %isIteratorLessThanEnd <- %i < %end

    cjump %isIteratorLessThanEnd = 1 :call_label_ciao_11
    goto :call_label_ciao_12
    :call_label_ciao_11
    %encodedI <- %i
    %encodedI <<= 1
    %encodedI <- %encodedI
    %encodedI += 1
    %encodedN <- %n
    %encodedN <<= 1
    %encodedN <- %encodedN
    %encodedN += 1
    rdi <- %encodedN
    rsi <- %encodedI

mem rsp -8 <- :call_label_ciao_13
    call @modulo 2

:call_label_ciao_13
    %remainder <- rax
    %i <- %i
    %i += 1
    %isDivisible <- %remainder = 1

    cjump %isDivisible = 1 :call_label_ciao_8
    goto :call_label_ciao_10
    :call_label_ciao_12
    rax <- 3
    return
  )
  (@modulo
    2
    %a <- rdi
    %b <- rsi
    %result <- %a
    %isBLessThanZero <- %b <= 1

    cjump %isBLessThanZero = 1 :call_label_ciao__ciao_14
    goto :call_label_ciao__ciao_15
    :call_label_ciao__ciao_14
    rax <- -1
    return
    :call_label_ciao__ciao_15
    %isALessThanB <- %a < %b

    cjump %isALessThanB = 1 :call_label_ciao__ciao_16
    goto :call_label_ciao__ciao_17
    :call_label_ciao__ciao_16
    rax <- %result
    return
    :call_label_ciao__ciao_17
    %result <- %result
    %result -= 1
    %result <- %result
    %result >>= 1
    %b <- %b
    %b -= 1
    %b <- %b
    %b >>= 1
    %result <- %result
    %result -= %b
    %result <- %result
    %result <<= 1
    %result <- %result
    %result += 1
    %b <- %b
    %b <<= 1
    %b <- %b
    %b += 1
    %isResultLessThanB <- %result < %b

    cjump %isResultLessThanB = 1 :call_label_ciao__ciao_16
    goto :call_label_ciao__ciao_17
  )
  (@main
    0
    rdi <- 11

mem rsp -8 <- :call_label_ciao__ciao__ciao_18
    call @firstNPrimes 1

:call_label_ciao__ciao__ciao_18
    %primes <- rax
    rdi <- %primes
    call print 1
    rdi <- 41

mem rsp -8 <- :call_label_ciao__ciao__ciao_19
    call @firstNPrimes 1

:call_label_ciao__ciao__ciao_19
    %primes <- rax
    rdi <- %primes
    call print 1
    rdi <- 103

mem rsp -8 <- :call_label_ciao__ciao__ciao_20
    call @firstNPrimes 1

:call_label_ciao__ciao__ciao_20
    %primes <- rax
    rdi <- %primes
    call print 1
    rdi <- 129

mem rsp -8 <- :call_label_ciao__ciao__ciao_21
    call @firstNPrimes 1

:call_label_ciao__ciao__ciao_21
    %primes <- rax
    rdi <- %primes
    call print 1
    rdi <- 1

mem rsp -8 <- :call_label_ciao__ciao__ciao_22
    call @firstNPrimes 1

:call_label_ciao__ciao__ciao_22
    %primes <- rax
    rdi <- %primes
    call print 1
    rdi <- -5

mem rsp -8 <- :call_label_ciao__ciao__ciao_23
    call @firstNPrimes 1

:call_label_ciao__ciao__ciao_23
    %primes <- rax
    rdi <- %primes
    call print 1
    rdi <- 3

mem rsp -8 <- :call_label_ciao__ciao__ciao_24
    call @firstNPrimes 1

:call_label_ciao__ciao__ciao_24
    %primes <- rax
    rdi <- %primes
    call print 1
    return
  )
)
