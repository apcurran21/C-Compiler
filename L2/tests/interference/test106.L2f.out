%array %arraySize %i r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%arraySize %array %i r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%i %array %arraySize r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r10 %array %arraySize %i r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r11 %array %arraySize %i r10 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r12 %array %arraySize %i r10 r11 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r13 %array %arraySize %i r10 r11 r12 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r14 %array %arraySize %i r10 r11 r12 r13 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r15 %array %arraySize %i r10 r11 r12 r13 r14 r8 r9 rax rbp rbx rcx rdi rdx rsi
r8 %array %arraySize %i r10 r11 r12 r13 r14 r15 r9 rax rbp rbx rcx rdi rdx rsi
r9 %array %arraySize %i r10 r11 r12 r13 r14 r15 r8 rax rbp rbx rcx rdi rdx rsi
rax %array %arraySize %i r10 r11 r12 r13 r14 r15 r8 r9 rbp rbx rcx rdi rdx rsi
rbp %array %arraySize %i r10 r11 r12 r13 r14 r15 r8 r9 rax rbx rcx rdi rdx rsi
rbx %array %arraySize %i r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rcx rdi rdx rsi
rcx %array %arraySize %i r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rdi rdx rsi
rdi %array %arraySize %i r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdx rsi
rdx %array %arraySize %i r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rsi
rsi %array %arraySize %i r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx
